### PR TITLE
Assorted coverity fixes

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

**Review by mrc0mmand — APPROVED** [review]

File: src/core/cgroup.c [hunk 1]
```diff
@@ -4573,6 +4573,7 @@ static uint64_t unit_get_effective_limit_one(Unit *u, CGroupLimitType type) {
                 }
 
         cc = unit_get_cgroup_context(u);
+        assert(cc);
```
Comment (YHNdnzj) [inline src/core/cgroup.c]: assert_se?

**Review by YHNdnzj — APPROVED** [review]

File: src/core/cgroup.c (hunk 1 above)
Comment (mrc0mmand) [reply src/core/cgroup.c]: It's not a test code and the expression has no side effects, so regular `assert()` is appropriate here.

File: src/core/cgroup.c [hunk 2]
```diff
@@ -4573,6 +4573,7 @@ static uint64_t unit_get_effective_limit_one(Unit *u, CGroupLimitType type) {
                 }
 
         cc = unit_get_cgroup_context(u);
```
Comment (poettering) [inline src/core/cgroup.c]: ASSERT_PTR()?

File: src/core/cgroup.c (hunk 1 above)
Comment (YHNdnzj) [reply src/core/cgroup.c]: (I meant to merge with the line above)

But yeah, maybe `ASSERT_PTR` is better.

Comment (bluca) [reply src/core/cgroup.c]: Nitpickers, assemble!

**Review by yuwata — APPROVED** [review]

**Pull request merged**